opennms-doc/guide-admin/src/asciidoc/text/plugins/opennms-es-rest-mapping.adoc

```diff
@@ -0,0 +1,2295 @@
+The id of each event is included in the _source` `id` field and also duplicated in the internal `_id`.
+
+Events in the _{opennms-product-name}_ events table (i.e. those corresponding to logs or taps) are copied directly to the `opennms-events-raw-`
+indexes. In _{opennms-product-name}_ alarms and events can contain `parameters` which  are name value pairs referencing additional data stored when the 
+event or alarm is created. In Elastic Search these parameters are always stored in separate fields in the index with names beginning with `p_`
+
+Alarm change events created by the _Alarm Change Notifier Plugin_ have an identical format to raw events 
+but are only copied to the `opennms-events-alarmchange-` indexes. These alarm change events are also used to change 
+the state of alarms in the `opennms-alarms-` indexes. Thus alarm entries in the `opennms-alarms-` indexes reflect the current state of 
+alarms as notified by _{opennms-product-name}_ through alarm change events. 
+
+The parameters included with each type of Alarm Change Event are listed in the _Alarm Change Notifier Plugin_ section. Each parameter in the index
+will have a p_ prefix. (Thus %parm[newalarmvalues]% becomes p_newalarmvalues).
+
+Alarms and Events have `severity` fields defined as integers (long) and also corresponding `severity_text` fields which give the 
+text equivalent (Critical,Major,Minor,Normal,Cleared etc.).
+
+===== Additional Alarm Fields
+
+The id of each alarm is included in the `_source` `alarmid` field and also duplicated in the internal `_id` reference for the alarms index. 
+Alarm Change Events reference their associated alarm using the `p_alarmid` parameter. So to make it easier to search for alarm change events associated
+with the same alarm, alarms also have a `_source`  `p_alarmid` parameter which matches `alarmid`.  Thus we should be able to search for an alarm in the `opennms-alarms` index
+and find its complete lifecycle from alarm raise to deletion in the `opennms-events-alarmchange` index
+
+The alarms index is enriched with additional data to allow the alarm entries to be used in SLA calculations.
+
+[options="header, autowidth"]
+|===
+| Additional Alarm Fields  | description 
+| alarmackduration | Calculated time in ms from first event which created the alarm to the latest alarm acknowledgement
+| alarmclearduration| Calculated time in ms from first event which created the alarm to the latest alarm clear
+| initialseverity | The final state of any given alarm in an alarm index should be cleared and deleted. Therefore we also include an initial severity
+| initialseverity_text | The initial severity as a text field
+|===
+
+
+
+
+
+===== Table of Index Mapping
+
+The following table describes the mapping of simple
+_{opennms-product-name}_ events to the Raw Events Index and the mapping of Alarm
+Change Events to the Alarm Change Events index and  to the Alarms
+index
+++++
+
+<table width="786" cellpadding="2" cellspacing="0">
```

Comment:
So the intention is to give a mapping which a user can follow which shows how OpenNMSs events end up in the elastic search indexes. This was asked for by the lead customer. After a lot of thought, the table seemed the easiest way to explain what is going on.  It is complex.
 I originally tried to do this in asciidoc but it was way too complicated to do the editing. In the end I resorted to editing the doc in libre office and generating the html for inclusion. If there was an easy way to either convert the libre office doc directly to asciidoc or editing the asciidoc table as easily as in office, I would gladly do it. Believe me I have tried.
Can you comment on the usefulness and readability of the generated documentation.